braindecode/datautil/serialization.py

```diff
@@ -6,17 +6,129 @@
 #
 # License: BSD (3-clause)
 
-import json
 import os
-from glob import glob
+import json
 import warnings
+from glob import glob
 
 import mne
+from mne.io.fiff import raw
 import pandas as pd
 
 from ..datasets.base import BaseDataset, BaseConcatDataset, WindowsDataset
 
 
+def create_fname_id_from_description(description):
+    # XXX: Enforce 'subject', 'recording' and/or 'run' in description?
+    fname = list()
+    if 'subject' in description:
+        fname.append(f'subj{description["subject"]}')
+    if 'recording' in description:
+        fname.append(f'rec{description["recording"]}')
+    if 'run' in description:
+        fname.append(f'run{description["run"]}')
+
+    if not fname:
+        raise ValueError(
+            'description must contain any of "subject", "recording" or "run".')
+
+    return '_'.join(fname)
+
+
+def save_base_dataset(dataset, path, overwrite=False):
```

Comment:
I would not expose anything else than the save method in BaseConcatDataset 

basically this needs to be private at best